### PR TITLE
chat how bad do you think i broke things on a scale of 1-10

### PR DESCRIPTION
**Pull request by NathanCrazy**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1746,13 +1746,15 @@ def battlelevel(levels, id, home=False):
...
+    except Exception:
+        latency = "infinite"
+    await message.response.send_message(f"cat has brain delay of {latency} ms " + str(get_emoji("staring_cat")))
+    # sorry i made it ugly :(
+    try:
         if latency >= 100:
             await achemb(message, "infinite", "send")
     except Exception:
-        latency = "infinite"
         await achemb(message, "infinite", "send")
```
Comment (milenakos) [inline main.py:1757]: can probably do with without except, something like `if latency == "infinite" or latency >= 100:`

File: main.py [hunk 2]
```diff
@@ -2666,59 +2666,23 @@ async def callback_hell(interaction, thing):
...
-            buttons_list.append(Button(label="Hidden", style=ButtonStyle.blurple))
-        lambdas_list.append(lambda interaction : (await callback_hell(interaction, "Hidden") for _ in '_').__anext__())
-        buttons_list[-1].callback = lambdas_list[-1]
+        for i in ["Cat Hunt", "Random", "Silly", "Hard", "Hidden"]: # hear me out what if it works :picardia_shrug:
+            if category == i:
+                buttons_list.append(Button(label=i, style=ButtonStyle.green))
+            else:
+                buttons_list.append(Button(label=i, style=ButtonStyle.blurple))
+            lambdas_list.append(lambda interaction : (await callback_hell(interaction, i) for _ in '_').__anext__())
+            buttons_list[-1].callback = lambdas_list[-1]
```
Comment (milenakos) [inline main.py:2685]: how about no???

**Review by milenakos — COMMENTED** [review]
:silence:

**Comment by NathanCrazy** [comment]
oh i just straight up removed achiever whoops

**Comment by milenakos** [comment]
> oh i just straight up removed achiever whoops

we love nathancrazy prs

**Comment by NathanCrazy** [comment]
truly are the best

File: main.py (hunk 2 above)
Comment (NathanCrazy) [reply main.py:2685]: awh its been sitting like that in the code for- hold on lemme check

**Comment by milenakos** [comment]
also you ate a few newlines but im gonna wait until you fix the other two things

File: main.py (hunk 2 above)
Comment (NathanCrazy) [reply main.py:2685]: github says yesterday but thats just because github thought you changed it yesterday because it loves being broken

Comment (milenakos) [reply main.py:2685]: that comment sat there since about the time /achievements was added

Comment (milenakos) [reply main.py:2685]: and i currently have no reason to believe it will magically work after years with no changes to it

Comment (milenakos) [reply main.py:2685]: yk usually old stuff breaks over time not in reverse

Comment (NathanCrazy) [reply main.py]: no i did change it lol

Comment (NathanCrazy) [reply main.py]: before it didn't have a defer in it so it wasn't going to work

File: main.py [hunk 3]
```diff
@@ -1746,13 +1746,11 @@ def battlelevel(levels, id, home=False):
...
         latency = round(bot.latency * 1000)
-        if latency >= 100:
-            await achemb(message, "infinite", "send")
     except Exception:
         latency = "infinite"
-        await achemb(message, "infinite", "send")
     await message.response.send_message(f"cat has brain delay of {latency} ms " + str(get_emoji("staring_cat")))
-
+    if latency == "infinite" or latency >= 100:
+        await achemb(message, "infinite", "send")
```
Comment (milenakos) [inline main.py:1754]: please uneat the newline

File: main.py [hunk 4]
```diff
@@ -2725,7 +2712,8 @@ async def callback_hell(interaction, thing):
         return myview
 
     await message.response.send_message(embed=gen_new("Cat Hunt"), ephemeral=True, view=insane_view_generator("Cat Hunt"))
-
+    if unlocked >= 15:
+        await achemb(message, "achiever", "send")
```
Comment (milenakos) [inline main.py:2717]: newline 2

File: main.py [hunk 5]
```diff
@@ -2925,6 +2914,8 @@ async def lb_handler(interaction, type, do_edit=None):
         except Exception:
             await interaction.followup.send(embed=embedVar, view=myview)
 
+        if leader:
+            await achemb(message, "leader", "send")
     # helpers! everybody loves helpers.
     async def slowlb(interaction):
```
Comment (milenakos) [inline main.py:2920]: newline 3

File: main.py [hunk 6]
```diff
@@ -2666,25 +2662,16 @@ async def callback_hell(interaction, thing):
...
 
-        # would be much more optimized but i cant get this to work
-        # for i in ["Cat Hunt", "Random", "Unfair"]:
-        #   if category == i:
-        #        buttons_list.append(Button(label=i, style=ButtonStyle.green))
-        #   else:
-        #        buttons_list.append(Button(label=i, style=ButtonStyle.blurple))
-        #   lambdas_list.append(lambda interaction : (await interaction.edit_original_response(embed=gen_new(i), view=insane_view_generator(i)) for _ in '_').__anext__())
-        #   buttons_list[-1].callback = lambdas_list[-1]
-
```
Comment (milenakos) [inline main.py:2687]: probably should leave the comment there idk :silence:

Comment (NathanCrazy) [reply main.py:2687]: can i at least put my updated one that might work there instead ig

**Comment by NathanCrazy** [comment]
okay i put the code that has a very good chance of working back into a comment

**Comment by milenakos** [comment]
> can i at least put my updated one that might work there instead ig

can we get something more reliable than "might work"? :staring_cat:

**Comment by NathanCrazy** [comment]
no i literally cant i dont even have an ide on this laptop im typing this in github web

**Comment by milenakos** [comment]
:staring_cat:

**Comment by NathanCrazy** [comment]
i mean throw it in cat bot 2 electric boogaloo ill test it

**Comment by milenakos** [comment]
my git bugged out some days ago so i had to nuke cat bot 2 :staring_cat:

**Comment by NathanCrazy** [comment]
:storing cat:
cant you just like ctrl c ctrl v change token throw it into the vps

**Comment by milenakos** [comment]
well ig lets try anyways

**Comment by NathanCrazy** [comment]
yooo okay ill uncomment it

**Comment by milenakos** [comment]
testing results: all buttons redirect to Hidden

**Comment by NathanCrazy** [comment]
how the fuck

**Comment by NathanCrazy** [comment]
wait no i can fix it ez

**Comment by milenakos** [comment]
it seems like all lambdas are the same

**Comment by NathanCrazy** [comment]
i can not fix it ez i was wrong i need to keep thinking

**Comment by milenakos** [comment]
we can probably simplify that to just `buttons_list[-1].callback = callback_hell` if we find a way to extract which button was clicked inside of the callback_hell itself

**Comment by NathanCrazy** [comment]
oh okay ill see if i can set that up

**Comment by milenakos** [comment]
maybe setting custom_id for buttons? that should help cus you can get them using interaction.data["custom_id"]

**Comment by NathanCrazy** [comment]
oh i didnt see that comment lol
shouldn't this work though? lol

**Comment by milenakos** [comment]
no not in the slightest it doesnt even match the function signature

**Comment by milenakos** [comment]
i think the issue is the fact i doesnt stay overtime and can probably be solved with i.copy() inside the lambda but my solution seems to work well enough

**Comment by milenakos** [comment]
im gonna push it rn

**Review by milenakos — APPROVED** [review]
lgtm

**Comment by MonkeysWithPie** [comment]
> im typing this in github web
> 
> 

are you using the github.dev or just the base "edit in place" thing

**Comment by NathanCrazy** [comment]
> > im typing this in github web
> 
> are you using the github.dev or just the base "edit in place" thing

edit in place lmao

File: main.py [hunk 7]
```diff
@@ -2745,7 +2756,7 @@ async def catch(message: discord.Interaction, msg: discord.Message):
         await achemb(message, "not_like_that", "send")
 
 
-# pointLaugh lives on in our memories
+# dementia
```
Comment (milenakos) [inline main.py:2723]: btw i did notice this but whatever screw you

**Pull request merged**